app/controllers/application_controller.rb

```diff
@@ -2,4 +2,24 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_filter :api_authentication
+
+  protected
+  def render_api_unauthorized
+    render json: "Invalid access token", status: :unauthorized
+  end
+
+  def api_authentication
+    verify_api_access_token || render_api_unauthorized
+  end
+
+  def verify_api_access_token
+    authenticate_with_http_token do |access_token, options|
```

Comment:
What do you mean by that? This is being used to validate access, but a custom json response is returned.